File: bench/kernels/bench_fib.nim
# bench_fib.nim - recursive Fibonacci, n=38
import std/[monotimes, times, strformat]

proc fib(n: int): int =
  if n <= 1: return n
  return fib(n - 1) + fib(n - 2)

let n = 38
let t0 = getMonoTime()
let result = fib(n)
let elapsed = int((getMonoTime() - t0).inMilliseconds)
echo &"result: {result}"
echo &"elapsed: {elapsed} ms"

File: bench/kernels/bench_fib.php
<?php
// bench_fib.php - recursive Fibonacci, n=38
function fib(int $n): int { return $n <= 1 ? $n : fib($n - 1) + fib($n - 2); }

$t0 = microtime(true);
$result = fib(38);
$elapsed = (int)((microtime(true) - $t0) * 1000);
echo "result: $result\n";
echo "elapsed: $elapsed ms\n";
